### DISCUSSION TITLE
Reading GeoJSON into geodataframe with non utf-8 encoding throws an error

### DISCUSSION THREAD

**Discussion by awa5114** [geopandas/geopandas#3477] [Q&A]
Is it currently possible to read a simple GeoJSON file hich has non-utf8 encoding for certain fields names and/or field values? Here's an example:

```
{
"type": "FeatureCollection",
"name": "SP07___SP22_BACHES_A_EAUX",
"features": [
{ "type": "Feature", "properties": { "OBJECTID": 1, "Id": 0, "N": 5, "Nom": "SP09", "Fon´┐¢ier": " ", "Phase": "Phase 1", "Date_": "15/06/23", "retenue": "", "PROJET_EN_COURS": "EXISTANTE" }, "geometry": { "type": "Point", "coordinates": [ 254448.773599997162819, 119554.822999998927116 ] } },
{ "type": "Feature", "properties": { "OBJECTID": 16, "Id": 0, "N": 3, "Nom": "SP07", "Fon´┐¢ier": " ", "Phase": "Phase 1", "Date_": "15/06/23", "retenue": "", "PROJET_EN_COURS": "EXISTANTE" }, "geometry": { "type": "Point", "coordinates": [ 254424.692699998617172, 121807.375199999660254 ] } }
]
}
```

The characters are recognized by notepad++ and appear as follows:

<img width="94" alt="image" src="https://github.com/user-attachments/assets/70424bcb-1431-4881-be70-8da143d6c2c1">

When I try to read the above file with `gpd.read_file("test.geojson")` I get an error:

> UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe7 in position 3: invalid continuation byte

Specifying the optional argument `encoding="utf-8"` or `encoding="latin1"` doesn't seem to help. Is this currently supported?

**Comment by theroggy** [comment]
The (Geo)JSON standard is quite strict in the encodings that are supported: only UTF-8, UTF-16 and UTF-32 ([JSON standard](https://www.rfc-editor.org/rfc/rfc7159#section-8.1)). Most JSON parsers also limit support to these encodings, including the libraries used by geopandas to read/write files.

But, as it is a text format, this is not really a problem. It is easy to let python do the conversion from `latin` to `UTF-8`: when you read the file, the result will be automatically converted to a `UTF-8` string. Next, `read_file` can then just read this string, like this:
``` python
import geopandas as gpd

path = r"C:\Temp\geojson_latin\latin.geojson"
with open(path, 'r', encoding='latin') as f:
    data = f.read()
    gdf = gpd.read_file(data)

print(gdf.columns)
```
Output:
```
Index(['OBJECTID', 'Id', 'N', 'Nom', 'Fonçier', 'Phase', 'Date_', 'retenue',
       'PROJET_EN_COURS', 'geometry'],
      dtype='object')
```
